### PR TITLE
Archive role check for queried Users State

### PR DESCRIPTION
**Pull request by heyrandhir**
- [x] Adds check for archive role before sending the list of queried users
- [x] Adds tests for the same

### CODE REVIEW

**Review by RitikJaiswal75 — COMMENTED** [review]
Can we have a test for it to try to store a user with a role archive and call the API and check for returned value?

**Review by manish591 — COMMENTED** [review]
LGTM

**Comment by heyrandhir** [comment]
> Can we have a test for it to try to store a user with a role archive and call the API and check for returned value?

Yes done 🫡

File: controllers/userStatus.js [hunk 1]
```diff
@@ -70,17 +70,21 @@ const getUserStatus = async (req, res) => {
...
+        status.username = result.user.username;
+        activeUsers.push(status);
+      }
     }
     return res.json({
       message: "All User Status found successfully.",
-      totalUserStatus: allUserStatus.length,
-      allUserStatus,
+      totalUserStatus: activeUsers.length,
+      allUserStatus: activeUsers,
```
Comment (Maheima) [inline controllers/userStatus.js:87]: Good job handling it like this here. This way you saved the extra iterations of changing allUserStatus everywhere it is used

**Comment by heyrandhir** [comment]
> LGTM! Good work Randhir 🎉

Thanks @Maheima 🙂

File: controllers/userStatus.js (hunk 1 above)
Comment (heyrandhir) [reply controllers/userStatus.js:87]: Yes changing it in backend was better than handling in frontend everywhere.

File: test/integration/userStatus.test.js [hunk 2]
```diff
@@ -63,6 +63,44 @@ describe("UserStatus", function () {
...
+      const response = await chai.request(app).get("/users/status?state=IDLE");
+      expect(response).to.have.status(200);
+      expect(response.body.message).to.equal("All User Status found successfully.");
+      expect(response.body.totalUserStatus).to.be.a("number");
+      expect(response.body.totalUserStatus).to.equal(1);
+      expect(response.body.allUserStatus).to.be.a("array");
+      expect(response.body.allUserStatus.length).to.equal(1);
+    });
+
+    it("Should only get the idle userStatus only for active user when both archived and active users are there in DB", async function () {
```
Comment (RitikJaiswal75) [inline test/integration/userStatus.test.js:91]: can we change the statement to `should return only non-archived user`

the above test for `should not get archived user` can be removed as this test checks for both

**Review by RitikJaiswal75 — APPROVED** [review]
LGTM, 
Thankyou for always being open to changes and implementing them in a jiffy.

**Review by manish591 — APPROVED** [review]

**Review by shmbajaj — APPROVED** [review]

**Pull request merged**